migrations/educandu-2023-02-01-02-change-cdn-urls.js

```diff
@@ -0,0 +1,197 @@
+import GithubFlavoredMarkdown from '../src/common/github-flavored-markdown.js';
+
+const OLD_ROOMS_CDN_PREFIX = 'cdn://rooms/';
+const OLD_MEDIA_CDN_PREFIX = 'cdn://media/';
+
+export default class Educandu_2023_02_01_02_change_cdn_urls {
+  constructor(db) {
+    this.db = db;
+    this.gfm = new GithubFlavoredMarkdown();
+  }
+
+  redactUrl(url = '') {
+    if (url.startsWith(OLD_ROOMS_CDN_PREFIX)) {
+      return url
+        .replace(OLD_ROOMS_CDN_PREFIX, 'cdn://room-media/')
+        .replace('/media/', '/');
+    }
+    if (url.startsWith(OLD_MEDIA_CDN_PREFIX)) {
+      return url.replace(OLD_MEDIA_CDN_PREFIX, 'cdn://document-media/');
+    }
+    return url;
+  }
+
+  redactMarkdown(text = '') {
+    return this.gfm.redactCdnResources(text, url => this.redactUrl(url));
+  }
+
+  updateDoc(doc) {
+    doc.cdnResources = doc.cdnResources.forEach(this.redactUrl);
```

Comment:
I think this should be `map` instead of `forEach` :thinking:

Reply:
😳 Thank you! Corrected.